scipy/stats/_discrete_distns.py

```diff
@@ -937,6 +937,21 @@ def _stats(self, a):
     def _entropy(self, a):
         return a / sinh(a) - log(tanh(a/2.0))
 
+    def _rvs(self, a):
+        # The discrete Laplace is equivalent to the two-sided geometric
+        # distribution with PMF:
+        #   f(k) = (1 - alpha)/(1 + alpha) * alpha^abs(k)
+        # Furthermore, the two-sided geometric distribution is
+        # equivalent to the difference between two iid geometric 
+        # distributions.
+        # Thus, we can leverage the following:
+        #   1) alpha = e^-a
+        #   2) probability_of_success = 1 - alpha (Bernoulli trial)
```

Comment:
Should these comments be moved up to the 'Notes' section of the [docstring](https://github.com/scipy/scipy/blob/3b6b8679e27397c5c3f61e6d150f331621990287/scipy/stats/_discrete_distns.py#L896-L907)?

Reply:
IMO either is OK.

Reply:
I added these comments so anyone looking at rvs() could see why the code works, especially for any future maintenance. But, I can also expand the docstring and examples to show users how this can be "transformed" into a geometric distribution by setting `a=-np.log(alpha)`.

BTW, where are the "before notes," "after notes," and "examples" pulled from?

Reply:
Expanding the docstring is best left to a follow-up PR. This is can be merged quickly once two small comments are addressed.

For how docstrings are assembled see the `__init__` methods of rv_discrete, it calls _construct_doc

Reply:
I added references to the comments.